### DISCUSSION TITLE
Improve SFTP Browsing

### DISCUSSION THREAD

**Discussion by 0x729000** [isontheline/pro.webssh.net#442] [Ideas]
![ED30B094-4B07-4B39-B517-39463D226CB2](https://user-images.githubusercontent.com/92766418/137827266-ec2cd995-5789-4667-a467-2530aa5f241a.jpeg)
Whilst browsing SFTP the experience is very limited. You may only open one file at a time and are unable to swipe through the files. It would be really nice to be able to open a dir and select the first file to then swipe left:right through the files.

May i also request you implement a thumbnail generator/display for viewing static image files?

**Comment by isontheline** [comment]
Hello @0x729000 👋

1. You can open multiple SFTP browsers instead : https://webssh.net/documentation/help/howtos/launching-multiple-terminals/
2. "Unable to swipe through the files" : Sorry I don't understand this statement. Could you explain?
3. The thumbnail generator could be technically implemented but, if you have big files it will slow down your file browsing experience. JPEG files could have thumbnails at the start of the file but not all and, not all file types.

**Reply by 0x729000** [reply]
Thanks for the quick response! 

To clarify, I suggest the SFTP browser allows you to scroll through previewable files such as images once one is opened. The current behaviour is that you open one file, and to view another you have to close the view, then tap the file, then view, to see the file. It is my view that the behaviour should be more like the iOS gallery app, whereby on opening a file/image, you can swipe through in a carousel.

**Reply by 0x729000** [reply]
Would it be possible to have the thumbnail generation as an option in the settings? A user configurable cache size could negate the slow loading after the first rendering.

**Reply by isontheline** [reply]
I now understand the wanted behavior 😇
Thank you a lot for your feedback 🙏

I'm creating a new issue with your ideas :  https://github.com/isontheline/pro.webssh.net/issues/445
